Mathlib/SetTheory/Cardinal/Basic.lean

```diff
@@ -1910,8 +1909,11 @@ theorem mk_iUnion_le {α ι : Type u} (f : ι → Set α) : #(⋃ i, f i) ≤ #
 
 theorem mk_iUnion_le_lift {α : Type u} {ι : Type v} (f : ι → Set α) :
     lift.{v} #(⋃ i, f i) ≤ lift.{u} #ι * ⨆ i, lift.{v} #(f i) := by
-  refine mk_iUnion_le_sum_mk_lift.trans <| Eq.trans_le ?_ (sum_le_iSup_lift _)
-  rw [← lift_sum, lift_id'.{_,u}]
+  apply mk_iUnion_le_sum_mk_lift.trans
+  convert sum_le_iSup_lift fun i ↦ lift.{v} #(f i)
+  · rw [← lift_sum, lift_id']
+  · exact lift_umax.symm
+  · rw [lift_lift]
```

Comment:
The fact that this lemma gets so much worse is quite a bad sign for your new lemma: it suggests that the proposed change is a decrease in usability. It seems to me that both the old and new versions of the lemma should exist, to avoid making users do the `lift` shuffling that you need to do here.

Reply:
I'm still just getting this to build! I'll analyze the situation once I've got a green checkmark.

Reply:
Yeah, I have to agree the new lemma is a bit inconvenient at times, mostly because `rw` can't see into the `iSup`. I've kept the old version and renamed the new one.